Small documentation updates (#25)
Aliucord/documentation: plugin-dev/6_finding_discord_stuff.md

# Finding Discord Stuff

To make any plugin that modifies discord classes and methods, you need to find what method does what and in which class it is. There are two ways to do so:

If you are comfortable with decompiling the entire discord source code, you can use JADX with the following command:

```
jadx --export-gradle --show-bad-code --fs-case-sensitive --respect-bytecode-access-modifiers --no-inline-methods --no-inline-anonymous --no-inline-kotlin-lambda --use-source-name-as-class-name-alias if-better --output-dir decompiled base.apk
```

> You can add `--no-generate-kotlin-metadata` if you do not want big kotlin metadata walls.

Reviewer: `--no-generate-kotlin-metadata` is only available in Aliucord's version of [JADX](https://github.com/Aliucord/jadx)

Author: I added clarification.

# Finding Discord Stuff

To make any plugin that modifies discord classes and methods, you need to find what method does what and in which class it is. There are two ways to do so:

If you are comfortable with decompiling the entire discord source code, you can use JADX with the following command:

```

> jadx --export-gradle --show-bad-code --fs-case-sensitive --respect-bytecode-access-modifiers --no-inline-methods --no-inline-anonymous --no-inline-kotlin-lambda --use-source-name-as-class-name-alias if-better --output-dir decompiled base.apk

Reviewer 2: ```suggestion
jadx --export-gradle --show-bad-code --fs-case-sensitive --respect-bytecode-access-modifiers --no-inline-methods --no-inline-anonymous --no-inline-kotlin-lambda --output-dir decompiled base.apk
```

`--use-source-name-as-class-name-alias` is a bad idea

also what is `if-better`?

Author: from the original repo:
`'if-better' - use source name if it seems better than the current one`

```

> If you use [Juby210's fork of JADX](https://github.com/Aliucord/jadx) you can add `--no-generate-kotlin-metadata` if you do not want big kotlin metadata walls.

Reviewer 2: ```suggestion
If you use Aliucord's [fork](https://github.com/Aliucord/jadx) of JADX you can add the `--no-generate-kotlin-metadata` flag to remove the `@Metadata` annotations on most classes.
```

> Either way, you can check [RazerTexz's repo](https://github.com/RazerTexz/Discord-JADX) to see the Discord source code.

Reviewer 2: I don't actually like this decompilation, it uses `--use-source-name-as-class-name-alias`

Author: i only included it since its more updated than Juby's repo but i can remove

> You can use the Developer Assistant app for finding view names while running Aliucord.

Reviewer 2: ```suggestion
You can use the [Developer Assistant](https://play.google.com/store/apps/details?id=com.appsisle.developerassistant&hl=en_US) app
to find View ids while running Aliucord.
```
